src/System.Private.Uri/src/System/Uri.cs

```diff
@@ -2186,7 +2194,8 @@ private unsafe ParsingError PrivateParseMinimal()
 
                     // This will disallow '\' as the host terminator for any scheme that is not implicitFile or cannot have a Dos Path
                     if ((idx < (ushort)length && pUriString[idx] == '\\') && NotAny(Flags.ImplicitFile) &&
-                        _syntax.NotAny(UriSyntaxFlags.AllowDOSPath))
+                        (_syntax.NotAny(UriSyntaxFlags.AllowDOSPath) ||
+                        (InFact(Flags.UncPath) && _syntax.NotAny(UriSyntaxFlags.ConvertPathSlashes))))
```

Comment:
It'll report error in cases where the authority ends with '\' on Unix. Since we don;t convert backward slash to forward slash on Unix, we expect the authority terminator to always be '/'